python/ray/dag/compiled_dag_node.py

```diff
@@ -1016,6 +1097,20 @@ def execute(
 
         self._get_or_compile()
 
+        if self._execution_index - self._max_execution_index > self._capacity:
+            if self._info_level_when_execution_may_block == "ERROR":
+                raise ValueError(
+                    "Execution may be blocked due to DAG is at its capacity "
+                    f"({self._capacity}). Please call ray.get() on previous "
+                    "CompiledDAGRefs to retrieve and release results from the DAG."
+                )
+            elif self._info_level_when_execution_may_block == "WARN":
+                logger.warn(
+                    "Execution may be blocked due to DAG is at its capacity "
```

Comment:
Same message here. Want to avoid the term capacity because I don't think it's clear what that means (e.g., could be memory capacity or number of executions).